ui/app/AppLayouts/Wallet/views/walletconnect/WalletConnectSDK.qml

```diff
@@ -176,32 +210,117 @@ WebView {
         }
     }
 
+    Timer {
+        id: eventsTimer
+
+        interval: 100
+        repeat: true
+        running: false
+
+        onTriggered: {
+            root.runJavaScript("window.wcEventResult ? window.wcEventResult.shift() : null", function(event) {
+                if (event) {
+                    switch(event.name) {
+                        case "session_request":
+                            root.sessionRequestEvent(event.payload)
+                            break
+                        default:
+                            console.error("WC unknown event type: ", event.type)
+                            break
+                    }
+                }
+            })
+        }
+    }
+
     QtObject {
         id: d
 
         property var sessionProposal: null
         property var sessionType: null
         property bool sdkReady: false
 
+        property ListModel pairingsModel: pairings
+
+        onSdkReadyChanged: {
+            if (sdkReady) {
+                d.getPairings()
+            }
+        }
+
+        function populatePairingsModel(pairList) {
+            pairings.clear();
+            for (let i = 0; i < pairList.length; i++) {
+                pairings.append({
+                    active: pairList[i].active,
+                    topic: pairList[i].topic,
+                    expiry: pairList[i].expiry
+                });
+            }
+        }
+
+
         function isWaitingForSdk() {
             return timer.running
         }
 
-
         function generateSdkCall(methodName, paramsStr, successState, errorState) {
-            return "wcResult = {error: null}; try { wc." + methodName  + "(" + paramsStr + ").then((callRes) => { wcResult = {state: " + successState + ", error: null, result: callRes}; }).catch((error) => { wcResult = {state: " + errorState + ", error: error}; }); } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message}; }; wcResult"
+            return "wcResult = {}; try { wc." + methodName  + "(" + paramsStr + ").then((callRes) => { wcResult = {state: " + successState + ", error: null, result: callRes}; }).catch((error) => { wcResult = {state: " + errorState + ", error: error}; }); } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message}; }; wcResult"
+        }
+        function requestSdkAsync(jsCode) {
+            root.runJavaScript(jsCode,
+                function(result) {
+                    timer.restart()
```

Comment:
@stefandunca why don't we create and call for example "handleResponse(result)" function from here instead of running the timer? Guess that the `result` here is the same as `wcResult` used in `onTriggered` slot of the timer?

Reply:
This is due to the polling mechanism that we use to track the result of async operations running inside `WebView`. The handler here is returning after the async operation was started and timer will check regularly for the answer then stop. I'm working to improve this part and make it more robust. Still, the only option we have now is to communicate between QML runtime and WebView runtime without creating a client/server approach which is unsafe.